File: demo/src/main/java/exercise1/UnauthorizeException.java
package exercise1;

import org.springframework.web.bind.annotation.ResponseStatus;
import org.springframework.http.HttpStatus;

@ResponseStatus(code = HttpStatus.UNAUTHORIZED) // ERROR 500 CODE
public class UnauthorizeException extends RuntimeException {
	private static final long serialVersionUID = -4383687061470914751L;

	public UnauthorizeException() {
	}

	public UnauthorizeException(String message) {
		super(message);
	}

	public UnauthorizeException(Throwable cause) {
		super(cause);
	}

	public UnauthorizeException(String message, Throwable cause) {
		super(message, cause);
	}

}

File: New folder/Demo - Kotlin/src/main/kotlin/demo/UnauthorizeException.kt
package demo

import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.ResponseStatus


@ResponseStatus(code = HttpStatus.UNAUTHORIZED) // ERROR 500 CODE
class UnauthorizeException : RuntimeException {
    constructor() {}
    constructor(message: String?) : super(message) {}
    constructor(cause: Throwable?) : super(cause) {}
    constructor(message: String?, cause: Throwable?) : super(message, cause) {}

    companion object {
        private const val serialVersionUID = -4383687061470914751L
    }
}
